pokemon_battle_arena/src/db/pokemon_token.rs

```diff
@@ -131,33 +117,42 @@ impl PokemonToken {
     /// Getter function for move one. If the move is set, the function returns it, if not,
     /// it returns None
     pub fn get_move_one(self) -> Option<moves::Technique> {
-        Some(self.move_one.unwrap_or(return None).0)
+        if let Some(x) = self.move_one {
+            return Some(x.0);
+        }
+        None
```

Comment:
As I told you, you can use `Option::map()` here. Please try to understand those helper methods. Those 4 lines can be replaced by `self.move_one.map(|x| x.0)`.